src/pages/DelegationCreatePage/components/DelegationCreateForm.tsx

```diff
@@ -0,0 +1,262 @@
+import { EServiceQueries } from '@/api/eservice/eservice.queries'
+import { useQuery } from '@tanstack/react-query'
+import {
+  CompactOrganization,
+  EServiceMode,
+  EServiceTechnology,
+  ProducerEService,
+} from '@/api/api.generatedTypes'
+import { useAutocompleteTextInput } from '@pagopa/interop-fe-commons'
+import { SectionContainer } from '@/components/layout/containers'
+import { useDialog } from '@/stores'
+import { TenantQueries } from '@/api/tenant'
+
+export type DelegationCreateFormValues = {
+  eserviceName: string
+  eserviceDescription: string
+  delegateId: string
+}
+
+export type EServiceCreateDraftValues = {
+  name: string
+  description: string
+  technology: EServiceTechnology
+  mode: EServiceMode
+}
+
+type DelegationCreateFormProps = {
+  delegationKind: 'CONSUME' | 'PROVIDE'
+  setActiveStep: React.Dispatch<React.SetStateAction<'KIND' | 'FORM'>>
+}
+
+const defaultValues: DelegationCreateFormValues = {
+  eserviceName: '',
+  eserviceDescription: '',
+  delegateId: '',
+}
+
+export const DelegationCreateForm: React.FC<DelegationCreateFormProps> = ({
+  delegationKind,
+  setActiveStep,
+}) => {
+  const { t } = useTranslation('party')
+
+  const [isChecked, setIsChecked] = useState(false)
+
+  const { openDialog } = useDialog()
+
+  const formMethods = useForm({ defaultValues })
+
+  const onSubmit = async (formValues: DelegationCreateFormValues) => {
+    if (!isChecked && delegationKind === 'PROVIDE') {
+      // if it is a producer delegation and isChecked is false the eservice must be created
+      const eserviceParams: EServiceCreateDraftValues = {
+        name: formValues.eserviceName,
+        description: formValues.eserviceDescription,
+        technology: 'REST',
+        mode: 'DELIVER',
+      }
+      openDialog({
+        type: 'delegations',
+        eserviceParams: eserviceParams,
+        delegationParams: { delegateId: formValues.delegateId },
+      })
+    } else {
+      const delegationParams = {
+        eserviceId: formValues.eserviceName,
+        delegateId: formValues.delegateId,
+      }
+      openDialog({
+        type: 'delegations',
+        delegationParams: delegationParams,
+      })
+    }
+  }
+
+  const [eserviceAutocompleteTextInput, setEserviceAutocompleteTextInput] =
+    useAutocompleteTextInput()
+
+  const selectedEServiceRef = React.useRef<ProducerEService | undefined>(undefined)
+
+  const formatAutocompleteOptionLabelEservice = React.useCallback(
+    (eservice: ProducerEService) => {
+      return `${eservice.name}`
+    },
+    [t]
+  )
+
+  /**
+   * TEMP: This is a workaround to avoid the "q" param in the query to be equal to the selected eservice name.
+   */
+  function getQ() {
+    let result = eserviceAutocompleteTextInput
+
+    if (
+      selectedEServiceRef.current &&
+      eserviceAutocompleteTextInput ===
+        formatAutocompleteOptionLabelEservice(selectedEServiceRef.current)
+    ) {
+      result = ''
+    }
+
+    return result
+  }
```

Comment:
Are you sure we need this here? The e-service cannot be pre-selected, there should be no need